lang/java/reef-bridge-client/src/main/java/org/apache/reef/bridge/client/AzureBatchBootstrapREEFLauncher.java

```diff
@@ -95,20 +87,24 @@ public static void main(final String[] args) throws IOException, InjectionExcept
         TANG.newInjector(generateConfiguration(jobSubmissionParameters))
             .getInstance(AzureBatchBootstrapDriverConfigGenerator.class);
 
+    final LocalAddressProvider defaultLocalAddressProvider =
+        Tang.Factory.getTang().newInjector().getInstance(LocalAddressProvider.class);
```

Comment:
>LocalAddressProvider [](start = 57, length = 20)

Concrete class breaks injection decoupling. Can you explain why this is being done?

Reply:
LocalAddressProvider is not a concrete class in java. It is the name of the interface.

---
In reply to: [201086560](https://github.com/sharathmalladi/reef/pull/5#discussion_r201086560) [](ancestors = 201086560)